e2e/app/element/base-element.ts

```diff
@@ -7,69 +8,69 @@ import {ClickOptions, ElementHandle, Page, WaitForSelectorOptions} from 'puppete
 export default class BaseElement {
 
   static asBaseElement(page: Page, elem: ElementHandle): BaseElement {
-    return new BaseElement(page, elem);
+    const baseElement = new BaseElement({puppeteerPage: page});
+    baseElement.element = elem;
+    return baseElement;
   }
 
-  protected readonly page: Page;
-  protected css: string;
-  protected xpath: string;
-  protected element: ElementHandle;
+  protected _element: ElementHandle;
+  protected readonly _xpath;
+  protected readonly _page: Page;
+  protected readonly _container: Container;
```

Comment:
Use a leading underscore `_` for class fields names. Main reason for change naming convention is because old class field `page` can be easily confused with jest-puppeteer global `page` property.

Reply:
The issue I see with this convention is that if you have two levels of parent classes, you'll need two leading underscores. But you don't know, when defining the parent class, at what level in the hierarchy it will wind up. It could be several.

@petesantos is this the usual convention? I see this answer as indicating underscores serve to make fields [effectively private](https://stackoverflow.com/questions/45167199/why-are-underscores-added-to-fields-of-a-typescript-class-in-angular4), which is a different consideration.

Reply:
The _ prefix in JS was a convention that was meant to indicate a member was private.

I am a bit confused about the intention of this naming convention. Is it intended to clearly separate these variables from globals? Or to clearly identify where in the prototype chain these are coming from? If it is for clarity I would prefer a more explicit naming convention.

Looking at the code without the context of this PR I would not know what the _ means unless it was an accepted convention within the code base.